.github/workflows/release.yaml

```diff
@@ -5,69 +5,128 @@ on:
     - name: checkout
-      uses: actions/checkout@v2
-    - name: tagname
-      id: tagname
-      run: |
-        echo releasing ${GITHUB_REF#refs\/tags/}
-        echo ::set-output name=tag::${GITHUB_REF#refs\/tags/}
+      uses: actions/checkout@v3
     - name: Set up QEMU for cross-building
       uses: docker/setup-qemu-action@v1
     - name: Set up Docker Buildx
       uses: docker/setup-buildx-action@v1
-    - name: Login to DockerHub
+
+    - name: Build image urls
+      id: image-urls
+      env:
+        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        QUAY_USERNAME: ${{ secrets.QUAY_USERNAME }}
+      run: |
+        IMAGES=ghcr.io/${IMAGE_NAME}
+        [[ -n "$DOCKER_USERNAME" ]] && IMAGES=${IMAGES},${IMAGE_NAME}
+        [[ -n "$QUAY_USERNAME" ]] && IMAGES=${IMAGES},quay.io/${IMAGE_NAME}
+        echo "::set-output name=images::${IMAGES}"
+
+    - name: Docker manager metadata
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: ${{ steps.image-urls.outputs.images }}
+        tags: |
+          type=ref,event=tag
+
+    - name: Log in to ghcr.io
       uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Log into DockerHub
+      uses: docker/login-action@v1
+      env:
+        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+      if: ${{ env.DOCKER_USERNAME != '' }}
       with:
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Log into quay.io
+      uses: docker/login-action@v1
+      env:
+        QUAY_USERNAME: ${{ secrets.QUAY_USERNAME }}
+      if: ${{ env.QUAY_USERNAME != '' }}
+      with:
+        registry: quay.io
+        username: ${{ secrets.QUAY_USERNAME }}
+        password: ${{ secrets.QUAY_PASSWORD }}
+
     - name: Build
       id: docker_build
       uses: docker/build-push-action@v2
+      env:
+        VERSION: ${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.version'] }}
       with:
-        platforms: linux/amd64,linux/arm64
+        context: .
         push: true
-        tags: |
-          ${{ format('{0}:{1}', env.IMAGE_NAME, steps.tagname.outputs.tag) }}
-          ${{ format('{0}:{1}', env.IMAGE_NAME, 'latest') }}
+        build-args: |
+          BINARY=cloud-provider-equinix-metal
+          LDFLAGS=-X 'k8s.io/component-base/version.gitVersion=${{ env.VERSION }}' -X 'k8s.io/component-base/version/verflag.programName=Cloud Provider Equinix Metal'
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+        platforms: linux/amd64,linux/arm64
+        cache-from: type=gha, scope=${{ github.workflow }}
+        cache-to: type=gha, mode=max, scope=${{ github.workflow }}
 
   release: # create a github actions release with necessary artifacts
     name: Release
     runs-on: ubuntu-latest
+    needs:
+    - images
```

Comment:
Add a dependency on the images job, so that the release is not present until the referenced image is created and pushed.